standalone/examples.go

```diff
@@ -0,0 +1,16 @@
+package standalone
+
+// Example model of an example gRPC request
+type Example struct {
+	Name    string `json:"name"`
+	Request struct {
+		Timeout  string `json:"timeout"`
+		Metadata []struct {
+			Name  string `json:"name"`
+			Value string `json:"value"`
+		} `json:"metadata"`
+		Data map[string]interface{} `json:"data"`
```

Comment:
This doesn't look right. Where did this come from?

This can actually be a primitive, like if the RPC request type is `google.protobuf.StringValue` or `google.protobuf.FloatValue`.

And it can be an array, if the method in question accepts a stream from the client (each element in the array is a message in the stream).

So I think we instead need `interface{}`. _However_, it is highly likely that someone constructing the example programmatically might want to construct an actual proto message and put it here. And proto messages do not correctly support marshaling via `encoding/json`. So I think this whole `Request` field needs to pulled out to a named type that has a custom `MarshalJSON` function, which will use `github.com/protobuf/jsonpb` to marshal the value(s) if it is a `proto.Message` or is a slice or array with `proto.Message` elements.

Reply:
I'll experiment with this.

Reply:
But even if you save the file and always unmarshal from JSON, if your RPC schema has a streaming request or one of the wrapper types as a request, this type won't work and instead needs to be `interface{}`.

I'm fine just having a TODO for now and leaving out the `jsonpb` stuff I suggested. But, to make sure I don't need to make any backwards incompatible changes to address that TODO, I think you need to hoist the `Request` field out to its own named type, instead of an anonymous struct type.

Personally, I can see creating examples programmatically being a very reasonable thing for someone to try upon seeing this API; and they'd be disappointed that it didn't work.